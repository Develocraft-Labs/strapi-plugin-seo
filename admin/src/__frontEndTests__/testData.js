import trads from "../translations";

export const id = 1,
  firstname = "John",
  lastname = "Doe",
  email = "John.Doe@companyA.com",
  title = "title",
  created_by = {
    id,
    email,
    firstname,
    lastname,
  },
  created_at = "2021-10-12T11:49:01.042Z",
  updated_at = "2021-10-15T09:43:36.344Z",
  updated_by = { email, firstname, id, isActive: true, lastname },
  seo = {
    canonical: "https://www.companyA.com/articles",
    created_at: "2021-10-12T11:55:00.595Z",
    created_by: 1,
    id: 1,
    jsonLd: {},
    metaDescription: "Artists Desc",
    metaTitle: "Artists Title",
    ogImage: "Artist og image",
    ogTitle: "Artists Og Title",
    ogType: "Artists Og Type",
    ogUrl: "Artists og url",
    title: "Artists Seo",
    updated_at,
    updated_by: 1,
  },
  resourceData = {
    title,
    id,
    created_by,
  },
  collectionName = "Artists",
  pagination = { page: 1, total: 13, pageSize: 5, pageCount: 3 },
  uid = "application::artists.artists",
  fullResults = [
    { created_at, created_by, id, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 2, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 3, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 4, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 5, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 6, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 7, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 8, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 9, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 10, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 11, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 12, seo, title, updated_at, updated_by },
    { created_at, created_by, id: 13, seo, title, updated_at, updated_by },
  ],
  seoId = 1,
  apiID = "artists",
  attributes = {
    seo: {
      dominant: false,
      nature: "oneWay",
      plugin: "seo",
      private: false,
      target: "plugins::seo.seo",
      unique: false,
    },
    title,
  },
  schema = {
    attributes,
    collectionName,
    connection: "default",
    description: "",
    draftAndPublish: true,
    kind: "collectionType",
    name: "Artists",
    pluginOptions: {},
    restrictRelationsTo: null,
    visible: true,
  },
  resourceSchema = { collectionName, schema },
  pluginUID = "plugins::seo.seo",
  projectCollectionTypes = [{ apiID, schema, uid }],
  selectedSeo = {
    canonical: "https://www.companyA.com/articles",
    created_at: "2021-10-12T11:55:00.595Z",
    created_by: created_by,
    id: 1,
    jsonLd: {},
    metaDescription: "Artists Desc",
    metaTitle: "Artists Title",
    ogImage: "Artist og image",
    ogTitle: "Artists Og Title",
    ogType: "Artists Og Type",
    ogUrl: "Artists og url",
    title: "Artists Seo",
    updated_at,
    updated_by: updated_by,
  },
  refreshOptions = () => {},
  metaTitle = "meta title",
  metaDescription = "meta description",
  canonical = "canonical",
  handleDeleteSeo = () => {},
  handleEditSeo = () => {},
  defaultLocale = "en",
  userEnabledLocales = [
    {
      code: "en",
      created_at: "2021-12-20T09:03:53.724Z",
      id: 1,
      isDefault: true,
      name: "English (en)",
      updated_at: "2021-12-20T09:03:53.724Z",
    },
  ],
  seos = [seo, fullResults];

export const resourceComponentTestData = {
  resourceData,
  resourceSchema,
};

export const seoFormComponentTestData = {
  pluginUID,
  seoId,
  projectCollectionTypes,
  resource: resourceComponentTestData,
};

export const contentTypesListComponentTestData = [
  {
    collectionName,
    fullResults,
    pagination,
    uid,
  },
];

export const translationPickerComponentTestData = {
  translations: trads,
  handleTranslation: () => {},
  value: "",
};

export const contentTypesItemComponentTestData = {
  uid,
  fullResults,
  pagination,
  collectionName,
};
export const previewComponentTestData = {
  selectedSeo,
  pluginUID,
  refreshOptions,
  projectCollectionTypes,
  handleDeleteSeo,
};

export const tableComponentTestData = {
  seos: { collectionName, fullResults, uid, pagination },
  handleDeleteSeo,
  handleEditSeo,
  userEnabledLocales,
  uid,
  defaultLocale,
};
