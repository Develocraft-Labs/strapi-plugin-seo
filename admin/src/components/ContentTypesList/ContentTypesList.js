import React, { useMemo } from "react";

import styled from "styled-components";
import Table from "../Table/Table";
import isValidLength from "../../utils/isValidLength";

const ListElement = styled.li`
  list-style: none;
  padding: 0rem 0rem 2rem 0rem;
`;

const List = styled.ul`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  padding: 0rem;
`;

/**
 * Component that renders a Content types in Strapi Project.
 * @returns {JSX.Element} List of either Single Types, Collection Types or both.
 */
const ContentTypesList = ({
  content,
  defaultLocale,
  userEnabledLocales,
  isSingleType,
}) => {
  if (!isValidLength(content)) {
    return null;
  }
  return (
    <List data-testid="content-types-list">
      {isSingleType ? (
        <SingleTypesListItems
          projectCollectionTypes={content}
          defaultLocale={defaultLocale}
          userEnabledLocales={userEnabledLocales}
          isSingleType={isSingleType}
        />
      ) : (
        <CollectionTypesListItems
          projectCollectionTypes={content}
          defaultLocale={defaultLocale}
          userEnabledLocales={userEnabledLocales}
        />
      )}
    </List>
  );
};

export const SingleTypesListItems = ({
  projectCollectionTypes,
  defaultLocale,
  userEnabledLocales,
  isSingleType,
}) => {
  const parsedData = useMemo(() => {
    const results = projectCollectionTypes.flatMap((singleType) => {
      const fullResults = singleType.fullResults;

      if (!isValidLength(fullResults)) {
        return [];
      }
      return fullResults.map((result) => ({
        ...result,
        uid: singleType.uid,
      }));
    });
    return { fullResults: results, collectionName: "Single Types" };
  }, [projectCollectionTypes]);

  if (!parsedData || !isValidLength(parsedData.fullResults)) {
    return null;
  }

  return (
    <ListElement>
      {userEnabledLocales ? (
        <Table
          seos={parsedData}
          userEnabledLocales={userEnabledLocales}
          uid={"placeholder"}
          defaultLocale={defaultLocale}
          isSingleType={isSingleType}
        />
      ) : null}
    </ListElement>
  );
};

const CollectionTypesListItem = ({
  uid,
  length,
  userEnabledLocales,
  items,
  defaultLocale,
}) => {
  return (
    <ListElement>
      {length > 0 && userEnabledLocales ? (
        <Table
          seos={items}
          userEnabledLocales={userEnabledLocales}
          uid={uid}
          defaultLocale={defaultLocale}
        />
      ) : null}
    </ListElement>
  );
};

const CollectionTypesListItems = ({
  projectCollectionTypes,
  defaultLocale,
  userEnabledLocales,
}) => {
  return (
    <>
      {projectCollectionTypes.map((items) => {
        const { uid } = items;
        const length = items?.fullResults?.length;
        if (!items) {
          return null;
        }
        return (
          <CollectionTypesListItem
            key={uid}
            uid={uid}
            length={length}
            userEnabledLocales={userEnabledLocales}
            items={items}
            defaultLocale={defaultLocale}
          />
        );
      })}
    </>
  );
};

export default ContentTypesList;
