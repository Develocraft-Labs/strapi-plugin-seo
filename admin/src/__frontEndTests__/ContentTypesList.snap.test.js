/**
 * @jest-environment jsdom
 */
import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";

import ContentTypesList from "../components/ContentTypesList/ContentTypesList";
import { contentTypesListComponentTestData } from "./testData";

// mock GlobalPagination component from strapi-helper-plugin
jest.mock("strapi-helper-plugin", () => {
  const Pagination = jest.fn(({ children }) => children);
  const GlobalPagination = jest.fn((props) => (
    <Pagination
      onChangeParams={props.handleParamsChange}
      params={props.params}
      count={props.count}
    >
      <div>1, 2</div>
    </Pagination>
  ));
  return {
    GlobalPagination: GlobalPagination,
    validateInput: () => [],
    InputErrors: () => null,
  };
});

describe("SNAPING TESTING CONTENTTYPELIST COMPONENT", () => {
  it("Should Render Successfully", () => {
    const component = renderer.create(
      <ContentTypesList content={contentTypesListComponentTestData} />
    );
    const tree = component.toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
    expect(tree.type).toMatch("ul");
  });
});
