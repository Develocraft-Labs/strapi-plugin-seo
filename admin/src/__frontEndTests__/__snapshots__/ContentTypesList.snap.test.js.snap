// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPING TESTING CONTENTTYPELIST COMPONENT Should Render Successfully 1`] = `
.c1 {
  list-style: none;
  padding: 0rem 0rem 2rem 0rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  padding: 0rem;
}

<ul
  className="c0"
  data-testid="content-types-list"
>
  <li
    className="c1"
  />
</ul>
`;
